src/Chapter2.hs

```diff
@@ -741,7 +755,9 @@ the list with only those lists that contain a passed element.
 
 🕯 HINT: Use the 'elem' function to check whether an element belongs to a list
 -}
-contains = error "contains: Not implemented!"
+contains ::  Int -> [[Int]] -> [[Int]]
+contains _ [] = []
+contains i list = filter (\x -> (filter (\y -> y == i) x) /= []) list
```

Comment:
You correctly used `elem` before. And now you need to combine `filter` and `elem` to implement this function using a combination of standard functions:

```suggestion
contains i list = filter (\xs -> elem i xs) list
```

After you remove the special case on the line above, you can eta-reduce this function as well. Here you have two opportunities for eta-reduction:

* The `list` argument to `filter`
* The `xs` argument to `elem`

Reply:
I see, I forgot completely the hint this time, I fixed it and even eta reduced the list arg, I think I am starting to get it!